src/commands/helper.py

```diff
@@ -1,27 +1,36 @@
+import sys
+from typing import Optional
 import discord
 from settings import get_settings
-from contextlib import asynccontextmanager
+from util import EmbedField, logging_embed_context
 
 settings = get_settings()
 
 
-@asynccontextmanager
-async def create_logging_embed(interaction: discord.Interaction):
-    embed = discord.Embed()
-    log_channel = interaction.guild.get_channel(settings.logging_channel_id)
-    try:
-        embed.set_author(
-            name=interaction.user.display_name,
-            icon_url=interaction.user.display_avatar.url,
-        )
-        embed.timestamp = interaction.created_at
-        embed.description = f"Used `{interaction.command.name}` command in {interaction.channel.mention}"
-        embed.add_field(name="Action", value=f"/{interaction.command.name}")
+def create_logging_embed(interaction: discord.Interaction, args: Optional[dict] = None):
```

Comment:
I think the way these methods are set up should be flipped. The commands should be interacting directly with the context manager via `async with logging_embed_context(...) as logging_embed:` and then `logging_embed_context` should be calling `create_logging_embed` to build the initial embed

Reply:
I'm afraid I don't follow. I intended to reuse `logging_embed_context(...)` in the autounexile log like so:
```python
def create_autounexile_embed(
    self, user: discord.User, exile_id: str, end_timestamp: str
):
    return logging_embed_context(
        self.get_channel(settings.logging_channel_id),
        user=user,
        timestamp=end_timestamp,
        description=f"<@{user.id}>'s exile has timed out",
        footer=f"Exile ID: {exile_id}",
    )
```

Reply:
Ah okay I see, that makes sense. Then let's rename this a little bit to still make it clear we're creating a context, something like `create_interaction_embed_context` and then the autounexile could be `create_autounexile_embed_context`